tests/src/core/testqgis.cpp

```diff
@@ -359,6 +360,21 @@ void TestQgis::testQgsRound()
   QGSCOMPARENEAR( qgsRound( 9999999.87654321987654321, 14 ), 9999999.876543219876543, 0.00000000000001 );
 }
 
+void TestQgis::testQgsVariantEqual()
+{
+  QVariant lhs;
+  QVariant rhs;
+
+  QVERIFY( lhs == rhs );
+  lhs.setValue( 0 );
+  QVERIFY( lhs != rhs );
+  rhs.setValue( 0 );
+  QVERIFY( lhs == rhs );
+  lhs.setValue( 1.2345 );
+  rhs.setValue( 1.2345 );
+  QVERIFY( lhs == rhs );
```

Comment:
Doest it work with `==`?

In any case, also add null variants (currently it's only testing invalid ones.

```
lhs = QVariant( QVariant::Int );
QVERIFY( !qgsVariantEqual( lhs, rhs ) );

rhs = QVariant( QVariant::Int );
QVERIFY( qgsVariantEqual( lhs, rhs ) )

```